Formula/libgraphqlparser.rb

```diff
@@ -15,11 +15,17 @@ class Libgraphqlparser < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on :macos # Due to Python 2
+
+  on_linux do
+    depends_on "pypy" => :build # Due to Python 2
+  end
```

Comment:
Not a huge fan of this (but not a very strong opinion).

I actually think we should consider deprecating formulae that still need Python2. Thoughts, @Homebrew/core?

Reply:
I'm open to opinions here as I'm mainly experimenting with this.

In terms of Python2 formulae, the most popular ones didn't work on quick attempt, i.e.
- `node@10` - failed with generic message. It is already deprecated, but is remains popular (at least on macOS = 5.5k+ installs/mo). It is also the dependency for `kibana`(deprecated)/`opensearch-dashboards`(should update to newer node that builds with Python3).
- `bazaar` - it builds, but has Python runtime dependency, with some code that is designed around CPython. It is a dependency for `influxdb`, which is reasonably popular (macOS = 2.4k+ installs/mo)

Reply:
I believe @iMichka and @sjackman have a expressed a strong opposition to supporting Python 2 in any way on Linux, and I completely agree with them on this.  I would support deprecating all formulae which still depend on Python 2 as well, unless the upstream team has provided a clear roadmap for when they will transition to Python 3.  We are coming up on 2 years since Python 2 EOL, and I think it is unacceptable for projects to depend on an unsupported language like this.

Specifically regarding `bazaar`, Fedora has replaced it with `breezy`, a maintained forked that should be drop in compatible: https://fedoraproject.org/wiki/Changes/ReplaceBazaarWithBreezy.  We could disable the `bazaar` formula and make `bzr` an alias to `breezy`.

For this particular formula, I think it should be deprecated and we should notify upstream that this has happened.  There has a been a pull request open for 3 years to add Python 3 support: https://github.com/graphql/libgraphqlparser/pull/66.  It was also noted in the issue that this repo may no longer have an active maintainer: https://github.com/graphql/libgraphqlparser/pull/66.

Reply:
Yeah, let's not add back python 2 support. So much work has been done to get rid of it.

Reply:
Agreed with deprecating these formulae.